File: app/src/main/java/com/gmail/wondergab12/restapi/dependencies/MainActivitySubGraph.java
package com.gmail.wondergab12.restapi.dependencies;

import androidx.appcompat.app.AppCompatActivity;

import com.gmail.wondergab12.restapi.MainActivity;

import dagger.BindsInstance;
import dagger.Subcomponent;

@Subcomponent(modules = MainActivityModule.class)
public interface MainActivitySubGraph {

    @Subcomponent.Builder
    interface Builder {

        @BindsInstance
        Builder with(AppCompatActivity activity);

        MainActivitySubGraph build();
    }

    void inject(MainActivity mainActivity);
}

File: app/src/main/java/com/gmail/wondergab12/restapi/dependencies/MainActivitySubGraph.kt
package com.gmail.wondergab12.restapi.dependencies

import androidx.appcompat.app.AppCompatActivity
import com.gmail.wondergab12.restapi.MainActivity
import dagger.BindsInstance
import dagger.Subcomponent

@Subcomponent(modules = [MainActivityModule::class])
interface MainActivitySubGraph {

    @Subcomponent.Builder
    interface Builder {

        @BindsInstance
        fun with(activity: AppCompatActivity): Builder

        fun build(): MainActivitySubGraph
    }

    fun inject(mainActivity: MainActivity)
}
